src/libLookup/Lookup.cpp

```diff
@@ -1939,22 +2046,38 @@ bool Lookup::ProcessSetLookupOnline(const vector<unsigned char>& message,
   }
 
   uint32_t portNo = 0;
-
-  if (!Messenger::GetLookupSetLookupOnline(message, offset, portNo)) {
+  PubKey lookupPubKey;
+  if (!Messenger::GetLookupSetLookupOnline(message, offset, portNo,
+                                           lookupPubKey)) {
     LOG_EPOCH(WARNING, to_string(m_mediator.m_currentEpochNum).c_str(),
               "Messenger::GetLookupSetLookupOnline failed.");
     return false;
   }
 
+  auto iter =
```

Comment:
Hmm this code block is repeated quite a few times. Maybe we can define it as a function somewhere.

Reply:
Good point, i will do it.

Reply:
I put the check into a function VerifyLookupNode, it will be more readable and easy to maintain.